src/hot-elements.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+export {};  // Ensure this file is a module.
+
+const processIsDebug =
+    typeof process === 'undefined' || process.env.NODE_ENV !== 'dev';
+const esmHmrIsPresent = !!(import.meta as HotImportMeta).hot;
+/**
+ * This code shouldn't be imported in prod, but if it is, try to give any
+ * bundler as many chances as it can to dead code eliminate it.
```

Comment:
From what I've seen, the DEC is often weak and I'm not sure it'll always propagate out of statements like this. You might have a better chance if this is in the if clause below. Over time it'd be good to document here that this was tested in which tools.